### PR TITLE
Fix(kclvm-runner): Added different file locks depending on the curren…

### PR DESCRIPTION
**Pull request by zong-zhe**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y 

issue #270.

#### 2. What is the scope of this PR (e.g. component or file name):

kclvm-runner

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [x] Other

Added different file locks depending on the currently package.
If the current package is the main package, the lock will lock a temporary file based on time and process id.
If the current package is not the main package, the lock will lock a file named 'package_path.ll.lock'.

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [x] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

#### 6. Release note

Please refer to [Release Notes Language Style Guide](https://kusionstack.io/docs/governance/release-policy/) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by Peefy — DISMISSED** [review]
LGTM

**Comment by ldxdl** [comment]
A quite basic and simple rule for any bug fix: you **MUST** write good tests for what you fix and prevent future errors.

File: kclvm/runner/src/assembler.rs [hunk 1]
```diff
@@ -300,10 +300,16 @@ impl KclvmAssembler {
         let prog_count = compile_progs.len();
         for (pkgpath, (compile_prog, import_names, cache_dir)) in compile_progs {
             let tx = tx.clone();
+            let entry_file = self.entry_file.clone();
```
Comment (Peefy) [inline kclvm/runner/src/assembler.rs:303]: Add more comments to explain it.

File: kclvm/runner/src/tests.rs [hunk 2]
```diff
@@ -32,13 +32,14 @@ const TEST_CASES: &[&str; 5] = &[
...
-const MULTI_FILE_TEST_CASES: &[&str; 6] = &[
-    "multi_file_compilation/no_kcl_mod_file",
-    "multi_file_compilation/relative_import",
-    "multi_file_compilation/relative_import_as",
-    "multi_file_compilation/import_abs_path/app-main",
-    "multi_file_compilation/import_regular_module",
-    "multi_file_compilation/import_regular_module_as",
+const MULTI_FILE_TEST_CASES: &[&str; 1] = &[
+    "multi_file_compilation/large_scale_case",
+    // "multi_file_compilation/no_kcl_mod_file",
```
Comment (Peefy) [inline kclvm/runner/src/tests.rs:37]: Why comment here.

File: kclvm/runner/src/test_datas/multi_file_compilation/large_scale_case/main.k [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import base.pkg.kusion_models.kube.frontend
```
Comment (Peefy) [inline kclvm/runner/src/test_datas/multi_file_compilation/large_scale_case/main.k:1]: The test suite is too large. Considering Konfig integration testing directly through git sub module

Comment (zong-zhe) [reply kclvm/runner/src/test_datas/multi_file_compilation/large_scale_case/main.k]: fixed

File: kclvm/runner/src/tests.rs (hunk 2 above)
Comment (zong-zhe) [reply kclvm/runner/src/tests.rs]: fixed

File: kclvm/runner/src/assembler.rs (hunk 1 above)
Comment (zong-zhe) [reply kclvm/runner/src/assembler.rs]: fixed

**Comment by ldxdl** [comment]
LGTM

**Review by Peefy — APPROVED** [review]
LGTM

**Pull request merged**